front/components/members/InvitationModal.tsx

```diff
@@ -0,0 +1,522 @@
+  newRole: RoleType;
+}) {
+  const r = await fetch(`/api/w/${owner.sId}/invitations/${invitation.sId}`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      status: invitation.status,
+      initialRole: newRole,
+    }),
+  });
+
+  if (!r.ok) {
+    const error: { error: { message: string } } = await r.json();
+    window.alert(error.error.message);
+  }
+  await mutate(`/api/w/${owner.sId}/invitations/${invitation.sId}`);
+}
+
+function ProPlanBillingNotice() {
+  return (
+    <ContentMessage size="md" variant="amber" title="Note">
+      <p>
+        New users will be charged a{" "}
+        <span className="font-semibold">
+          monthly fee of {getPriceWithCurrency(PRO_PLAN_29_COST)}
+        </span>
+        .{" "}
+      </p>
+      <br />
+      <p>
+        Next month's bill will be adjusted proportionally based on the members'
+        sign-up date.
+      </p>
+    </ContentMessage>
+  );
+}
+
+async function sendInvitations({
+  owner,
+  emails,
+  invitationRole,
+  sendNotification,
+  isNewInvitation,
+}: {
+  owner: WorkspaceType;
+  emails: string[];
+  invitationRole: ActiveRoleType;
+  sendNotification: any;
+  isNewInvitation: boolean;
+}) {
+  const body: PostInvitationRequestBody = emails.map((email) => ({
+    email,
+    role: invitationRole,
+  }));
+
+  const res = await fetch(`/api/w/${owner.sId}/invitations`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(body),
+  });
+
+  if (!res.ok) {
+    let data: any = {};
+    try {
+      data = await res.json();
+    } catch (e) {
+      // ignore
+    }
+    if (data?.error?.type === "invitation_already_sent_recently") {
+      sendNotification({
+        type: "error",
+        title: emails.length === 1 ? "Invite failed" : "Invites failed",
+        description:
+          (emails.length === 1 ? "This user has" : "These users have") +
+          " already been invited in the last 24 hours. Please wait before sending another invite.",
+      });
+    }
+
+    sendNotification({
+      type: "error",
+      title: "Invite failed",
+      description:
+        "Failed to invite new members to workspace: " + res.statusText,
+    });
+  } else {
+    const result: PostInvitationResponseBody = await res.json();
+    const failures = result.filter((r) => !r.success);
+
+    if (failures.length > 0) {
+      sendNotification({
+        type: "error",
+        title: "Some invites failed",
+        description:
+          result[0].error_message ||
+          `Failed to invite ${failures} new member(s) to workspace.`,
+      });
+    } else {
+      sendNotification({
+        type: "success",
+        title: "Invites sent",
+        description: isNewInvitation
+          ? `${emails.length} new invites sent.`
+          : `Sent ${emails.length} invites again.`,
+      });
+    }
+  }
+}
+
+export async function revokeInvitation({
```

Comment:
revokeInvitation and handleInvitationRoleChange are ~ the same function they should be the same

Rename `revokeInvitation` into `updateInvitation`
Keep the notifications but use the term "update" instead of "revoke"
Take the initialRole as argument and you can get rid of `handleInvitationRoleChange`